### PR TITLE
google-analytics

### PR DESCRIPTION
**Pull request by Hanuhitha**
Merge develop to main

### CODE REVIEW

File: libs/state/src/index.ts [hunk 1]
```diff
@@ -2,3 +2,4 @@ export * from './lib/contact/contact.actions';
 export * from './lib/contact/contact.state';
 export * as ResourceIds from './lib/resource-ids';
 export * from './lib/source-list/source-list.state';
+export { StateAnalyticsPluginService } from './lib/state-analytics-plugin/state-analytics-plugin.service';
```
Comment (axdanbol) [inline libs/state/src/index.ts:5]: Also needs to export the provider token

File: libs/state/src/lib/state-analytics-plugin/state-analytics-plugin.service.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+@Injectable({
+  providedIn: 'root',
+})
+export class StateAnalyticsPluginService implements NgxsPlugin {
+  /**
+   * Creates an instance of logger plugin.
+   * @param ga
+   */
+  constructor(private readonly ga: GoogleAnalyticsService) {}
```
Comment (axdanbol) [inline libs/state/src/lib/state-analytics-plugin/state-analytics-plugin.service.ts:25]: Use `inject` instead

File: libs/state/src/lib/state-analytics-plugin/state-analytics-plugin.service.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+  /**
+   * Handles logger plugin
+   * @param state
+   * @param action
+   * @param next
+   * @returns
+   */
+  handle(state: unknown, action: unknown, next: NgxsNextPluginFn) {
+    console.log('In handle');
```
Comment (axdanbol) [inline libs/state/src/lib/state-analytics-plugin/state-analytics-plugin.service.ts:35]: Remove logs

File: libs/state/src/lib/state-analytics-plugin/state-analytics-plugin.service.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  /**
+   * Logs action trigerred to google analytics event
+   * @param action
+   * @returns action
+   */
+  private logAction(action: unknown): void {
+    const type = getActionTypeFromInstance(action);
+    console.log('type', type);
+
+    if (type) {
```
Comment (axdanbol) [inline libs/state/src/lib/state-analytics-plugin/state-analytics-plugin.service.ts:49]: We should only log approved actions. Not all of them

File: libs/state/src/lib/state-analytics-plugin/state-analytics-plugin.service.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  }
+
+  /**
+   * Serializes action data
+   * @param this
+   * @param key
+   * @param value
+   * @returns serialize
+   */
+  private serialize(this: void, key: unknown, value: unknown): unknown {
```
Comment (axdanbol) [inline libs/state/src/lib/state-analytics-plugin/state-analytics-plugin.service.ts:63]: @bherr2 Should we update the serialization logic to include nested objects, etc.?

**Review by axdanbol — CHANGES_REQUESTED** [review]

File: libs/state/src/lib/state-analytics-plugin/state-analytics-plugin.service.ts (hunk 4 above)
Comment (axdanbol) [reply libs/state/src/lib/state-analytics-plugin/state-analytics-plugin.service.ts:47]: Not resolved. Type needs to be checked against approved action list/set to determine if the action should be logged

File: libs/state/src/lib/state-analytics-plugin/state-analytics-plugin.service.ts (hunk 2 above)
Comment (axdanbol) [reply libs/state/src/lib/state-analytics-plugin/state-analytics-plugin.service.ts:25]: Not resolved!

File: libs/state/src/lib/state-analytics-plugin/state-analytics-plugin.service.ts (hunk 5 above)
Comment (bherr2) [reply libs/state/src/lib/state-analytics-plugin/state-analytics-plugin.service.ts:60]: First argument to ga.event (See <https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#ec>) has a 150 byte limit, second has a 500 byte limit, 3rd also has a 500 byte limit). If it goes over, it gets truncated.

File: libs/state/src/lib/state-analytics-plugin/state-analytics-plugin.service.ts [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+});
+
+/**
+ * Google Analytics Plugin Injectable
+ */
+@Injectable({
+  providedIn: 'root',
+})
+
+/**
```
Comment (axdanbol) [inline libs/state/src/lib/state-analytics-plugin/state-analytics-plugin.service.ts:33]: Remove this comment. Class comment appears before decorator

File: libs/state/src/lib/state-analytics-plugin/state-analytics-plugin.service.ts [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+  /**
+   * Logs action trigerred to google analytics event
+   * @param action
+   * @returns action
+   */
+  private logAction(action: unknown): void {
+    const type = getActionTypeFromInstance(action);
+    if (type && this.loggableTypes.has(type)) {
+      console.log('Received Action', action);
```
Comment (axdanbol) [inline libs/state/src/lib/state-analytics-plugin/state-analytics-plugin.service.ts:68]: Remove log

**Review by axdanbol — CHANGES_REQUESTED** [review]
Increase test coverage to pass checks

File: libs/state/src/lib/state-analytics-plugin/state-analytics-plugin.service.ts [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+    return undefined;
+  }
+}
+
+/**
+ * State Login Provider
+ */
+export const STATE_LOGGER_PROVIDER: Provider = {
+  provide: NGXS_PLUGINS,
+  useClass: StateAnalyticsPluginService,
```
Comment (axdanbol) [inline libs/state/src/lib/state-analytics-plugin/state-analytics-plugin.service.ts:91]: Update this property to `useExisting` instead

**Review by axdanbol — CHANGES_REQUESTED** [review]

**Review by axdanbol — APPROVED** [review]

**Pull request merged**